### PR TITLE
LT-358: [Trading History] Add new endpoint to get open & executed orders report

### PR DESCRIPTION
**Pull request by lykke-vashetsin**

### CODE REVIEW

File: Scripts/dbo.OrdersHistory.sql [hunk 1]
```diff
@@ -65,4 +65,12 @@ IF NOT EXISTS(SELECT 'X'
 BEGIN
     ALTER TABLE [dbo].[OrdersHistory]
     ALTER COLUMN CorrelationId NVARCHAR(250) NULL;
+END;
+
+IF NOT EXISTS(SELECT 1 FROM sys.columns 
+          WHERE Name = N'CreatedBy'
+          AND Object_ID = Object_ID(N'[dbo].[OrdersHistory]'))
+BEGIN
+    alter table [nova].[dbo].[OrdersHistory] add CreatedBy as CASE WHEN ISJSON(AdditionalInfo) > 0 THEN CAST(json_value(AdditionalInfo, '$.CreatedBy') AS NVARCHAR(256)) ELSE null END
```
Comment (anokhin-lkk) [inline Scripts/dbo.OrdersHistory.sql:74]: How fast does it work? Maybe we just need to save this data explicitly?

File: src/MarginTrading.TradingHistory.SqlRepositories/OrdersHistorySqlRepository.cs [hunk 2]
```diff
@@ -205,12 +328,110 @@ private Task RollbackExceptionHandler(Exception exception)
...
+                .ToList();
+            var slList = tpSlList
+                .Where(x => x.Type == OrderType.StopLoss || x.Type == OrderType.TrailingStop)
+                .OrderByDescending(x => x.ModifiedTimestamp)
+                .ToList();
+
+            orderHistoryEntities.ForEach(x =>
+            {
+                var tp = tpList.FirstOrDefault(l =>
+                    l.ParentOrderId == x.Id && l.ModifiedTimestamp >= x.ModifiedTimestamp);
```
Comment (anokhin-lkk) [inline src/MarginTrading.TradingHistory.SqlRepositories/OrdersHistorySqlRepository.cs:383]: I am 99% sure this logic is not correct. Order could be modified after tp/sl order and condition l.ModifiedTimestamp >= x.ModifiedTimestamp is not needed here. But probably we should compare with relevance timestamp here.

File: src/MarginTrading.TradingHistory/Controllers/OrderBlotterController.cs [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        
+        private static OrderForOrderBlotterContract Convert(IOrderHistoryForOrderBlotterWithAdditionalData history)
+        {
+            var exchangeRate = history.FxRate == 0 ? 1 : 1 / history.FxRate;
+            decimal? notional = null;
+            decimal? notionalEUR = null;
+            if (history.Status == OrderStatus.Executed && history.ExecutionPrice.HasValue)
+            {
+                notional = Math.Abs(history.Volume * history.ExecutionPrice.Value);
+                notionalEUR = notional / exchangeRate;
```
Comment (anokhin-lkk) [inline src/MarginTrading.TradingHistory/Controllers/OrderBlotterController.cs:79]: Looks overcomplicated. I think you could just use * history.FxRate here

File: Scripts/dbo.OrdersHistory.sql (hunk 1 above)
Comment (anokhin-lkk) [inline Scripts/dbo.OrdersHistory.sql:74]: Are you sure that this computed field will work fast? Maybe we should just store this field explicitly in the broker?

File: src/MarginTrading.TradingHistory/Settings/ServiceSettings/DbSettings.cs [hunk 4]
```diff
@@ -19,5 +19,8 @@ public class DbSettings
 
         [Optional]
         public TimeSpan OrderHistoryForSupportExecutionTimeout { get; set; } = TimeSpan.FromMinutes(1);
+
+        [Optional]
+        public TimeSpan OrderBlotterExecutionTimeout { get; set; } = TimeSpan.FromMinutes(1);
```
Comment (anokhin-lkk) [inline src/MarginTrading.TradingHistory/Settings/ServiceSettings/DbSettings.cs:24]: Please make sure that this setting is mentioned in the task deployment instructions

File: src/MarginTrading.TradingHistory.SqlRepositories/OrdersHistorySqlRepository.cs [hunk 5]
```diff
@@ -99,26 +158,109 @@ public Task AddAsync(IOrderHistory order, ITrade trade)
...
+                    return nameof(IOrderHistoryForOrderBlotter.Volume);
+                case OrderBlotterSortingColumn.Validity:
+                    return nameof(IOrderHistoryForOrderBlotter.ValidityTime);
+                case OrderBlotterSortingColumn.CreatedOn:
+                    return nameof(IOrderHistoryForOrderBlotter.CreatedTimestamp);
+                case OrderBlotterSortingColumn.ExchangeRate:
+                    return nameof(IOrderHistoryForOrderBlotter.FxRate);
+                case OrderBlotterSortingColumn.ModifiedOn:
+                    return nameof(IOrderHistoryForOrderBlotter.ModifiedTimestamp);
+                default: throw new NotImplementedException();
```
Comment (tarurar) [inline src/MarginTrading.TradingHistory.SqlRepositories/OrdersHistorySqlRepository.cs:177]: Better to raise here `ArgumentOutOfRangeException `.

**Review by tarurar — APPROVED** [review]

**Pull request merged**